### ISSUE TITLE
Incorrect version constraints on importlib-metadata

### ISSUE THREAD

**Issue by tschoonj** [conda-forge/flake8-feedstock#52]
`flake8` has recently changed [the way](https://gitlab.com/pycqa/flake8/-/blob/master/setup.cfg) it depends on importlib-metadata:

```
install_requires=
# We document the reasoning for using ranges here:
# http://flake8.pycqa.org/en/latest/faq.html#why-does-flake8-use-ranges-for-its-dependencies
# And in which releases we will update those ranges here:
# http://flake8.pycqa.org/en/latest/internal/releases.html#releasing-flake8
    pyflakes >= 2.3.0, < 2.4.0
    pycodestyle >= 2.7.0, < 2.8.0
    mccabe >= 0.6.0, < 0.7.0
    importlib-metadata<4.3; python_version<"3.8"
```

This is affecting `pip check` invocations in recipes that depend on `flake8`, [such as molecule-docker,](https://github.com/conda-forge/molecule-docker-feedstock/pull/6) and is breaking builds.

Would it be possible to have a look at this? If necessary I will gladly open a PR myself. Note that this will mean getting rid of `noarch: python` 😢

**Comment by dopplershift** [comment]
Right, but for `importlib-metadata`, we generally just unconditionally install it to avoid losing noarch. I'm not sure what the problem is with changing to:
```yaml
  - importlib-metadata <4.3
```
We were already slightly inconsistent with upstream, which before they blocked newer versions of importlib-metadata had:
```ini
importlib-metadata; python_version<"3.8"
```
we ignore the python version so we can keep noarch, because it drastically reduces the number of builds.

**Comment by tschoonj** [comment]
Not a fan of this as you're introducing a dependency with an upper version limit that is not even going to be used by most.

But I will open a PR according to your suggestion.

**Comment by dopplershift** [comment]
@tschoonj I agree it's not ideal. But it's better than introducing builds across all Python versions (currently 3.7, 3.8, 3.9) across Windows, MacOS (both x86_64 and arm64), Linux (x86_64, aarch64, and ppc64le), and PyPy 3.7. This would then require rebuilding the flake8 package just to get 3.10 support (which has started as we speak.)
You're talking about *hours* of CI builds, for an otherwise pure Python package, in the name of saving users:
<img width="964" alt="image" src="https://user-images.githubusercontent.com/221526/139952653-33481648-4adc-43be-acec-fc023771f619.png">
3.7kb. 🤨 I wish noarch packages had the ability to do conditional dependencies, but alas.

IMO in this case "Practicality beats purity." You are of course welcome to bring this up to the rest of conda-forge.
